src/15-budget-intent/index.ts

```diff
@@ -0,0 +1,167 @@
+/**
+ * This experiment extracts a structured budget intent from user-supplied free text.
+ *
+ * Running this will generate multiple results across a number of examples and models.
+ *
+ * The results can be view in budgetIntents.ndjson
+ *
+ * They can also be converted to csv (via csv-kit) with:
+ *
+ * $ cat budgetIntents.ndjson | in2csv -f ndjson > budgetIntents.csv
+ *
+ * These^ csv results and some stats can be viewed at:
+ * https://docs.google.com/spreadsheets/d/1S1-7IJOPc26cgSlWaxP1bz9HEuNJXgQJTDdAyx7XRB0/edit?usp=sharing
+ *
+ */
+import _ from "lodash"
+import { anthropic } from "@ai-sdk/anthropic"
+import { LanguageModel, generateObject } from "ai"
+import { openai } from "@ai-sdk/openai"
+import { z } from "zod"
+import chalk from "chalk"
+import dedent from "dedent"
+import dotenv from "dotenv"
+import fs from "fs"
+import pAll from "p-all"
+import pSeries from "p-series"
```

Comment:
In order to fire off 30 async LLM calls in a way that didn't trip any API concurrency limits I had to orchestrate all the `Promise`'d results more carefully.

The promise-handling utils from https://github.com/sindresorhus/promise-fun are invaluable for that.

See L153, L156 below.

Reply:
👀 What a great library!